### PR TITLE
reuse InformerFactory in scheduler tests

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes # https://github.com/kubernetes/kubernetes/issues/107710

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/internal/queue/testing.go [hunk 1]
```diff
@@ -25,6 +25,11 @@ import (
 	"k8s.io/kubernetes/pkg/scheduler/framework"
 )
 
+// NewTestQueue creates a priority queue with an empty informer factory.
```
Comment (denkensk) [inline pkg/scheduler/internal/queue/testing.go:28]: Revert to the original place

File: pkg/scheduler/internal/queue/testing.go [hunk 2]
```diff
@@ -34,13 +39,17 @@ func NewTestQueueWithObjects(
...
 	informerFactory := informers.NewSharedInformerFactory(fake.NewSimpleClientset(objs...), 0)
+	return NewTestQueueWithInformerFactory(ctx, lessFn, informerFactory, opts...)
+}
+
+func NewTestQueueWithInformerFactory(
+	ctx context.Context,
+	lessFn framework.LessFunc,
+	informerFactory informers.SharedInformerFactory,
+	opts ...Option,
+) *PriorityQueue {
```
Comment (denkensk) [inline pkg/scheduler/internal/queue/testing.go:50]: The same informerFactory is started twice.

File: pkg/scheduler/scheduler_test.go [hunk 3]
```diff
@@ -916,6 +900,14 @@ func setupTestScheduler(ctx context.Context, queuedPodStore *clientcache.FIFO, s
 		recorder = &events.FakeRecorder{}
 	}
 
+	var schedulingQueue *queue.PriorityQueue
+	if informerFactory == nil {
+		informerFactory = informers.NewSharedInformerFactory(fake.NewSimpleClientset(), 0)
```
Comment (denkensk) [inline pkg/scheduler/scheduler_test.go:905]: I don't think we need to create informerFactory. if it is nil, let the test fail. WDYT?

File: pkg/scheduler/internal/queue/testing.go [hunk 4]
```diff
@@ -34,13 +39,17 @@ func NewTestQueueWithObjects(
 	opts ...Option,
 ) *PriorityQueue {
 	informerFactory := informers.NewSharedInformerFactory(fake.NewSimpleClientset(objs...), 0)
+	return NewTestQueueWithInformerFactory(ctx, lessFn, informerFactory, opts...)
```
Comment (Huang-Wei) [inline pkg/scheduler/internal/queue/testing.go:42]: I guess no callers need to pass in both informerFactory and objs?

File: pkg/scheduler/scheduler_test.go (hunk 3 above)
Comment (Huang-Wei) [reply pkg/scheduler/scheduler_test.go:905]: I think we can remove `informerFactory` from the function signature; instead, instantiate an informerFactory unconditionally here, and return it for the caller to consume.

(BTW: and check if we can eliminate `NewTestQueue`)

Comment (kerthcet) [reply pkg/scheduler/scheduler_test.go:905]: > I don't think we need to create informerFactory here. if it is nil, let the test fail. WDYT?

Aha, made a mistake here, it was not my really intension. Sorry.

File: pkg/scheduler/internal/queue/testing.go (hunk 1 above)
Comment (kerthcet) [reply pkg/scheduler/internal/queue/testing.go:28]: My intension here was to read the code from up to down.

File: pkg/scheduler/scheduler_test.go (hunk 3 above)
Comment (kerthcet) [reply pkg/scheduler/scheduler_test.go:902]: > I think we can remove `informerFactory` from the function signature; instead, instantiate an informerFactory unconditionally here, and return it for the caller to consume.

 We have some places build informer before: https://github.com/kubernetes/kubernetes/blob/834349ed1979073b4f6ed3e16393d31c19008701/pkg/scheduler/scheduler_test.go#L966-L977
> (BTW: and check if we can eliminate `NewTestQueue`)

`NewTestQueue` used at quite a lot of places and I think it simplified the usage for users to construct a priority queue, so I think we can keep it.

**Comment by alculquicondor** [comment]
Don't wait for my review on this one :)

**Comment by Huang-Wei** [comment]
/lgtm
/approve

Thanks for the cleanup @kerthcet .

**Pull request merged**